reporter/src/main/kotlin/reporters/evaluatedmodel/EvaluatedModelMapper.kt

```diff
@@ -445,13 +455,27 @@ internal class EvaluatedModelMapper(private val input: ReporterInput) {
                 issues += addIssues(this.issues, EvaluatedOrtIssueType.ANALYZER, dependency, null, packagePath)
             }
 
+            val children = visitDependencies { dependencies ->
+                dependencies.map { node ->
+                    if (!deduplicateDependencyTree || node.getInternalId() !in visitedNodes) {
+                        val childNode = node.toEvaluatedTreeNode(scope, path + dependency)
+                        visitedNodes += node.getInternalId() to childNode
+                        childNode
+                    } else {
+                        // If this node has already been visited, cut the subtree here. So only return the node
+                        // without its children.
+                        val duplicateRoot = visitedNodes.getValue(node.getInternalId())
+                        duplicateRoot.associateWithScope(scope)
+                        duplicateRoot.copy(children = emptyList())
+                    }
+                }.toList()
+            }
```

Comment:
As I understand the deduplication is intended to not work across scopes (from another comment on the PR of yours). The state of `visitedNodes` seems to be kept across scopes. Is there need to clear the map inbetween processing the scopes?

Reply:
Good catch! This has been fixed.

I also spotted one additional problem: Since the node was added to the `visitedNodes` map only after its children have been processed, it could be that duplicate nodes were missed. This has been changed as well, and the test has been adapted.